paasta_tools/marathon_tools.py

```diff
@@ -766,17 +776,42 @@ def deformat_job_id(job_id):
     return decompose_job_id(job_id)
 
 
-def read_namespace_for_service_instance(name, instance, cluster=None, soa_dir=DEFAULT_SOA_DIR):
-    """Retreive a service instance's nerve namespace from its configuration file.
-    If one is not defined in the config file, returns instance instead."""
+def read_all_namespaces_for_service_instance(name, instance, cluster=None, soa_dir=DEFAULT_SOA_DIR):
+    """Retreive all registration namespaces for a particular service instance.
+
+    For example, the 'main' paasta instance may register in the 'main'
+    namespace as well as the 'bulk' namespace
+
+    If one is not defined in the config file, returns a list containing
+    instance instead.
+    """
     if not cluster:
         cluster = load_system_paasta_config().get_cluster()
     srv_info = service_configuration_lib.read_extra_service_information(
         name,
         "marathon-%s" % cluster,
         soa_dir
     )[instance]
-    return srv_info['nerve_ns'] if 'nerve_ns' in srv_info else instance
+
+    registration_namespaces = srv_info.get('registration_namespaces', [])
+    # Backwards compatbility with nerve_ns
+    # FIXME(jlynch|2016-08-02, PAASTA-4964): DEPRECATE nerve_ns and remove it
+    if 'nerve_ns' in srv_info:
+        registration_namespaces.append(srv_info['nerve_ns'])
+
+    return registration_namespaces or [instance]
+
+
+def read_namespace_for_service_instance(name, instance, cluster=None, soa_dir=DEFAULT_SOA_DIR):
+    """Retreive a service instance's primary registration namespace for a
+    particular service instance.
+
+    This is the namespace that clients ought talk to, as well as paasta
+    ought monitor. In this context "primary" just means the first one in the
+    list of registration namespaces.
+
+    If one is not defined in the config file, returns instance instead."""
+    return read_all_namespaces_for_service_instance(name, instance, cluster, soa_dir)[0]
 
 
 def get_proxy_port_for_instance(name, instance, cluster=None, soa_dir=DEFAULT_SOA_DIR):
```

Comment:
can you update the docstring for this: "Get the proxy_port defined in the _first_ namespace configuration for a service instance."